Mathlib/RingTheory/Finiteness/Finsupp.lean

```diff
@@ -15,6 +16,40 @@ import Mathlib.LinearAlgebra.Quotient.Basic
 open Function (Surjective)
 open Finsupp
 
+namespace LinearMap
+
+variable {R M N ι : Type*} (S : Type*) [Semiring R] [AddCommMonoid M] [AddCommMonoid N]
+variable [Module R M] [Module R N] [Semiring S] [Module S N] [SMulCommClass R S N]
+
+/-- The linear map from `Hom(M,N)^(ι)` to `Hom(M,N^(ι))`. -/
```

Comment:
Could you mention that this is the finitely-supported version of `LinearEquiv.linearMapPi`?

Given that @kckennylau asked about the above on zulip, can I suggest spinning out a PR with just `.LinearEquiv.linearMapPi` and `.LinearEquiv.finsuppLinearMap`? The names still look a bit wonky to me, and I think a PR with just these results in would help get them right.

Reply:
I'm not sure I'll get back to this soon so @kckennylau please feel free to spin off the PR if you need this soonish.